src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -1,43 +1,131 @@
+const pMap = require("p-map");
+const pick = require("lodash/pick");
+const { removeEmpty } = require("../code-challenge/general");
+const { runTestEvaluator } = require("../code-challenge/test-evaluator");
 const { handleInternalLabel } = require("../../../../utils/general");
+const { compareIds } = require("../code-challenge/utils");
 
-// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
-async function runMetaTests(eventTests, eventMetaTests) {
+const conductMetaTestsOnTests = async (
+  eventTests,
+  eventMetaTests,
+  challengeLabel
+) => {
   const [internalTests, metaTests] = await Promise.all([
     getInternalTests(eventTests),
     getMetaTests(eventMetaTests),
   ]);
+  for (let i = 0; i < metaTests.length; i++) {
+    let metaTestResult = await executeTests(
+      metaTests[i].caseCode,
+      internalTests,
+      metaTests[i].label,
+      challengeLabel,
+      metaTests[i].id,
+      metaTests[i].passes
+    );
 
-  // console.log("final internalTests: ", internalTests);
-  // console.log("final metaTests: ", metaTests);
+    // MetaTest Final Results being logged to Console.
+    await console.log(
+      `\n\n<"${metaTests[i].label}" METATEST #${metaTests[i].id}> \n\n`,
+      metaTestResult,
+      "\n\n"
+    );
+  }
+};
 
-  // For next part of feature: return some type of eval using internalTests & metaTests
-  return false;
-}
+const executeTests = async (
+  metaCaseCode,
+  internalTests,
+  metaLabel,
+  challengeLabel,
+  metaTestId,
+  evalResultShouldBe
+) => {
+  return pMap(
+    internalTests?.filter(removeEmpty) || [],
+    async (internalTest) => {
+      const {
+        label: internalTestLabel,
+        internalTestCode: internalTestCode,
```

Comment:
you can just do `internalTestCode`, you don't need `internalTestCode: internalTestCode`

Reply:
Thanks. I changed it.